tests/common/util.rs

```diff
@@ -2219,24 +2358,29 @@ impl UChild {
     /// [`JoinHandle`]: std::thread::JoinHandle
     pub fn pipe_in<T: Into<Vec<u8>>>(&mut self, content: T) -> &mut Self {
         let ignore_stdin_write_error = self.ignore_stdin_write_error;
-        let content = content.into();
-        let stdin = self
-            .raw
-            .stdin
-            .take()
-            .expect("Could not pipe into child process. Was it set to Stdio::null()?");
-
-        let join_handle = thread::spawn(move || {
-            let mut writer = BufWriter::new(stdin);
-
-            match writer.write_all(&content).and_then(|()| writer.flush()) {
-                Err(error) if !ignore_stdin_write_error => Err(io::Error::new(
-                    io::ErrorKind::Other,
-                    format!("failed to write to stdin of child: {error}"),
-                )),
-                Ok(()) | Err(_) => Ok(()),
-            }
-        });
+        let mut content: Vec<u8> = content.into();
+        if self.stdin_pty.is_some() {
+            content.append(&mut END_OF_TRANSMISSION_SEQUENCE.to_vec());
+        }
+        let mut writer = self.take_stdin_as_writer();
+
+        let join_handle = std::thread::Builder::new()
+            .name("pipe_in".to_string())
+            .spawn(move || {
+                let result = writer.write_all(&content);
+                let result2 = result.and_then(|()| writer.flush());
```

Comment:
Is there a specific purpose here to creating 2 temporary result values ?

Reply:
Thanks for pointing out. I was debugging this part. Thats why I added the intermediate result values. But this can be removed. I'll do it.